src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestUpdater.cs

```diff
@@ -938,152 +940,102 @@ private TimeSpan GetReminderDelay(DateTime updatedAt)
     /// <summary>
     /// Alternative to ProcessPendingUpdatesAsync that is used in the code flow (VMR) scenario.
     /// </summary>
-    private async Task<bool> ProcessCodeFlowUpdateAsync(
+    private async Task ProcessCodeFlowUpdateAsync(
         SubscriptionUpdateWorkItem update,
         InProgressPullRequest? pr)
     {
-        bool isCodeFlow = update.SubscriptionType == SubscriptionType.DependenciesAndSources;
-        // Compare last SHA with the build SHA to see if we already have this SHA in the PR
         if (update.SourceSha == pr?.SourceSha)
         {
             _logger.LogInformation("PR {url} for {subscription} is already up to date ({sha})",
                 pr.Url,
                 update.SubscriptionId,
                 update.SourceSha);
 
-            await SetPullRequestCheckReminder(pr, isCodeFlow);
+            await SetPullRequestCheckReminder(pr, isCodeFlow:true);
             await _pullRequestUpdateReminders.UnsetReminderAsync(isCodeFlow: true);
-
-            return true;
+            return;
         }
 
-        // The E2E order of things for is:
-        // 1. We send a request to PCS and wait for a branch to be created. We note down this in the codeflow status. We set a reminder.
-        // 2. When reminder kicks in, we check if the branch is created. If not, we repeat the reminder.
-        // 3. When branch is created, we create the PR and set the usual reminder of watching a PR (common with the regular subscriptions).
-        // 4. For new updates, we only delegate those to PCS which will push in the branch.
-        if (pr == null)
-        {
-            (var targetRepository, var targetBranch) = await GetTargetAsync();
-
-            // Step 1. Create a branch for us
-            var prBranch = await CreateCodeFlowBranchAsync(update, targetBranch);
-            if (prBranch == null)
-            {
-                _logger.LogInformation("No changes required for subscription {subscriptionId}, no pull request created", update.SubscriptionId);
-                return true;
-            }
-
-            // Step 2. Create a PR
-            await CreateCodeFlowPullRequestAsync(
-                update,
-                targetRepository,
-                prBranch,
-                targetBranch);
+        var subscription = await _barClient.GetSubscriptionAsync(update.SubscriptionId);
+        var build = await _barClient.GetBuildAsync(update.BuildId);
+        var isForwardFlow = subscription.TargetDirectory != null;
+        string prHeadBranch = pr == null ? GetNewBranchName(subscription.TargetBranch) : pr.HeadBranch;
```

Comment:
This is a useful C# feature:
```suggestion
        string prHeadBranch = pr?.HeadBranch ?? GetNewBranchName(subscription.TargetBranch);
```